### ISSUE TITLE
Share extension is not opening the app when I have shared some content - IOS

### ISSUE THREAD

**Issue by rrios-dev** [Expensify/react-native-share-menu#112]
Hello, I have created an RN project with this [template](https://github.com/echobind/react-native-template), I was able to implement the share extension in Android and IOS, but in IOS when I try to share an URL from a website instead of open the app the share extension is opening the browser with this URL in my local machine, I have checked that groups are right in the IOS capabilities, and I think that the info.plist and appDelegate.m is correct because the share extension appears when I try to share content.

any idea?

**Comment by ahmadardal** [comment]
I am experiencing a similiar problem. I have followed the instructions and it is working on android. I can see my app in the Share Menu. But when I click on my app then it only opens the main page of my app and inserts the shared data. It doesn't open the custom screen which it should since I followed all the instructions. 

Even worse on iOS, I can see my app in the Share menu but when I click on my app nothing happens and no data is passed to my app either.

**Comment by krunal-lia** [comment]
@crystalii I'm facing the same issue on ios, no data is passed to my app. Not getting the callback from the listener

**Comment by qsdamar** [comment]
> Hello, I have created an RN project with this [template](https://github.com/echobind/react-native-template), I was able to implement the share extension in Android and IOS, but in IOS when I try to share an URL from a website instead of open the app the share extension is opening the browser with this URL in my local machine, I have checked that groups are right in the IOS capabilities, and I think that the info.plist and appDelegate.m is correct because the share extension appears when I try to share content.
> 
> any idea?

I'm having the same issue. Android works fine but on iOS share function opens a shared link on my local machine.

**Comment by qsdamar** [comment]
Fixed by deleting initial `ShareViewController.swift` from ShareExtension folder, and copying a new `ShareViewController.swift` file from `node-modules/react-native-share-menu/ios/ShareViewController.swift`.

That step has been explained in the docs, but I missed it previously.

<img width="1032" alt="Screenshot 2021-03-26 at 09 14 20" src="https://user-images.githubusercontent.com/56024932/112602487-0a67d880-8e14-11eb-9faf-e72fa7698836.png">

**Comment by rrios-dev** [comment]
> Fixed by deleting initial `ShareViewController.swift` from ShareExtension folder, and copying a new `ShareViewController.swift` file from `node-modules/react-native-share-menu/ios/ShareViewController.swift`.
> 
> That step has been explained in the docs, but I missed it previously.
> 
> <img alt="Screenshot 2021-03-26 at 09 14 20" width="1032" src="https://user-images.githubusercontent.com/56024932/112602487-0a67d880-8e14-11eb-9faf-e72fa7698836.png">

did you get the next error when you implemented the ShareViewController?
![Screenshot 2021-03-26 at 9 35 57 AM](https://user-images.githubusercontent.com/8117062/112604641-ac88c000-8e16-11eb-80b2-0ab552cc5415.png)

**Comment by qsdamar** [comment]
> did you get the next error when you implemented the ShareViewController?
> ![Screenshot 2021-03-26 at 9 35 57 AM](https://user-images.githubusercontent.com/8117062/112604641-ac88c000-8e16-11eb-80b2-0ab552cc5415.png)

Nope. After that, everything worked well. Perhaps you should check [this issue](https://github.com/meedan/react-native-share-menu/issues/103), it could be related.

**Comment by rrios-dev** [comment]
For the moment I wasn't able to build successfully in IOS, I have tried to import ShareViewController from react-native-share-menu, and also I have tried with a custom empty ShareViewController but I have the same problem related to the previous comment. Ping @caiosba in case that know what could happens

**Comment by ahmadardal** [comment]
Okay I have fixed the problem on my part. I created a new project by npx react-native init, and followed the instructions carefully. I can now share on my iPhone and I also receive the custom share menu view.

**Comment by dungdt-itealteam** [comment]
@crystalii can you help me check what's problem that i'm missing in config? Now i can share file and show custom share view. But when i press "continue in app" it bring me to my app but i'm not get any data in function ShareMenu.addNewShareListener) or ShareMenu.getInitialShare.

**Comment by nrazon** [comment]
I am experiencing a similar problem. I have followed the instructions and it is working on android and IOS OS version >16. 
But on OS 15.8.2 (iphone 7); I can see my app in the Share Menu. 
But when I click on my app  in the Share menu nothing happens and no data is passed to my app either.

Any Idea why it doesn't open my app on ios os 15.8.2 while showing my app on share menu.

**Comment by golu7679** [comment]
> I am experiencing a similar problem. I have followed the instructions and it is working on android and IOS OS version >16. But on OS 15.8.2 (iphone 7); I can see my app in the Share Menu. But when I click on my app in the Share menu nothing happens and no data is passed to my app either.
> 
> Any Idea why it doesn't open my app on ios os 15.8.2 while showing my app on share menu.

I am getting the same error, did you find any solution?
